kong/pdk/service/request.lua

```diff
@@ -284,6 +284,14 @@ local function new(self)
       ngx.var.upstream_host = value
     end
 
+    if string_lower(header) == ":authority"
+       and (ngx.var.upstream_scheme == "grpc" or
```

Comment:
Same, `ngx.var` can be safely cached as well. `local ngx_var = ngx.var`.

Question, if request is not gRPC and user attempts to set `:authority`, should we return an error instead? AFAIK there is no reason to set a plain-text header named `:authority` for non gRPC requests.